src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseAsyncMethodInAsyncContextTests.cs

```diff
@@ -1343,6 +1343,218 @@ public async Task<string> Foo()
             return VerifyCS.VerifyAnalyzerAsync(code);
         }
 
+        [Fact, WorkItem(7271, "https://github.com/dotnet/roslyn-analyzers/issues/7271")]
+        public Task WhenPassingZeroToSemaphoreSlimWait_NoDiagnostic()
+        {
+            const string code = """
+                                using System;
+                                using System.Threading;
+                                using System.Threading.Tasks;
+
+                                class Test
+                                {
+                                    async Task M()
+                                    {
+                                        SemaphoreSlim s = new SemaphoreSlim(0);
+                                        s.Wait(0);
```

Comment:
💭 I feel like all four of the "zero" cases could be in one `[Theory]`, but broken out is fine.